src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,53 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_LENGTH = 10;
+    private K[] key;
+    private V[] value;
+    private int storageSize;
+
+    public StorageImpl() {
+        key = (K[]) new Object[MAX_LENGTH];
+        value = (V[]) new Object[MAX_LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < storageSize; i++) {
+            if (compare(this.key[i], key)) {
+                this.value[i] = value;
+                return;
+            }
+        }
+        if (isStorageFull()) {
+            throw new RuntimeException("Storage is full! Can't add key: "
+                    + key + ", value: " + value);
+        }
+        this.key[storageSize] = key;
+        this.value[storageSize] = value;
+        storageSize++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < storageSize; i++) {
+            if (compare(this.key[i], key)) {
+                return this.value[i];
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return storageSize;
+    }
+
+    private boolean isStorageFull() {
+        return storageSize == MAX_LENGTH;
+    }
+
+    private boolean compare(Object obj1, Object obj2) {
```

Comment:
this method is used to compare keys, so let's parametrize and rename your 'keys' 
```suggestion
    private boolean compare(K key1, K key2) {
```